src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -166,12 +172,12 @@ trait SlickExpressionParser extends JavaTokenParsers {
   // the root function starts the pipeline at some root entity type in the workspace
   private def entityRootFunc(context: SlickExpressionContext): PipeType = {
     for {
-      entity <- entityQuery.findEntityByName(context.workspaceContext.workspaceId, context.rootType, context.rootName)
-    } yield entity
+      entity <- entityQuery.findEntities(context.workspaceContext.workspaceId, context.rootEntities.map(_.id))
+    } yield (entity.name, entity)
```

Comment:
can we put the entity's id in the temp table too then do a join instead of a possibly large inSet expression?